src/dmd/traits.d

```diff
@@ -1332,7 +1332,17 @@ extern (C++) Expression semanticTraits(TraitsExp e, Scope* sc)
             return 0;
         }
 
-        ScopeDsymbol._foreach(sc, sds.members, &pushIdentsDg);
+        __gshared int counter;
+        if (counter > 1000)
```

Comment:
How is this counter calculated, like why is it exactly when it's above 1000?

Reply:
Well, it's my noobish attempt at fixing this stack overflow and completely arbitrary. This function (`semanticTraits`) recursively calls itself and I bet there's a more sophisticated way in DMD to protect itself against such recursive overflows, but I couldn't find one.

Reply:
Actually, there isn't. Look how CTFE detects infinite recursion:
https://github.com/dlang/dmd/blob/master/src/dmd/dinterpret.d#L923
where CTFE_RECURSION_LIMIT is ... 1000.  

Or template instantiation: https://github.com/dlang/dmd/blob/master/src/dmd/dsymbolsem.d#L5640 

I guess this fix is fine for now, until a better mechanism is found.

Reply:
Would it make sense to leverage the same limit, rather than create a new one?

Reply:
I guess we could add a symbol recursionLimit which can be used throughout the compiler